djstripe/event_handlers.py

```diff
@@ -14,95 +15,195 @@
+def charge_webhook_handler(event, event_data, event_type, event_subtype):
+    """ Handles updates for charge objects. """
+    _handle_crud_type_event(Charge, event_data, event_subtype)
+
+
+@webhooks.handler("invoice")
 def invoice_webhook_handler(event, event_data, event_type, event_subtype):
-    versioned_invoice_data = Invoice(stripe_id=event_data["object"]["id"]).api_retrieve()
-    Invoice.sync_from_stripe_data(versioned_invoice_data)
+    """ Handles updates for invoice objects. """
+    _handle_crud_type_event(Invoice, event_data, event_subtype)
 
 
-# ---------------------------
-# InvoiceItem model events
-# ---------------------------
-@webhooks.handler(['invoiceitem'])
+@webhooks.handler("invoiceitem")
 def invoiceitem_webhook_handler(event, event_data, event_type, event_subtype):
-    versioned_invoiceitem_data = InvoiceItem(stripe_id=event_data["object"]["id"]).api_retrieve()
-    InvoiceItem.sync_from_stripe_data(versioned_invoiceitem_data)
+    """ Handles updates for invoice item objects. """
+    _handle_crud_type_event(InvoiceItem, event_data, event_subtype)
 
 
-# ---------------------------
-# Plan model events
-# ---------------------------
-@webhooks.handler(['plan'])
+@webhooks.handler("plan")
 def plan_webhook_handler(event, event_data, event_type, event_subtype):
-    versioned_plan_data = Plan(stripe_id=event_data["object"]["id"]).api_retrieve()
-    Plan.sync_from_stripe_data(versioned_plan_data)
+    """ Handles updates for plan objects. """
+    _handle_crud_type_event(Plan, event_data, event_subtype)
+
+
+#
+# Helpers
+#
+
+class CrudType(object):
+    """ Helper object to determine CRUD-like event state. """
+    created = False
+    updated = False
+    deleted = False
+
+    def __init__(self, **kwargs):
+        for k, v in six.iteritems(kwargs):
+            setattr(self, k, v)
+
+    @property
+    def valid(self):
+        """ Returns True if this is a CRUD-like event. """
+        return self.created or self.updated or self.deleted
+
+    @classmethod
+    def determine(cls, event_subtype, exact=False):
+        """
+        Determines if the event subtype is a crud_type (without the 'R') event.
+
+        :param event_subtype: The event subtype to examine.
+        :param exact: If True, match crud_type to event subtype string exactly.
+        :returns: The CrudType state object.
+        :rtype: ``CrudType``
+        """
+        crud_type = cls()
+
+        def check(crud_type_event):
+            if exact:
+                return event_subtype == crud_type_event
+            else:
+                return event_subtype.endswith(crud_type_event)
+
+        if check("updated"):
+            crud_type.updated = True
+        elif check("created"):
+            crud_type.created = True
+        elif check("deleted"):
+            crud_type.deleted = True
+
+        return crud_type
+
+
+def _handle_crud_type_event(
+        cls, event_data, event_subtype, stripe_id=None, customer=None,
```

Comment:
also, use ``target_cls`` instead of ``cls`` since this isn't, strictly speaking, a class method.